src/librustc_typeck/check/demand.rs

```diff
@@ -583,12 +624,25 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 }
                 (&ty::Float(ref exp), &ty::Float(ref found)) => {
                     if found.bit_width() < exp.bit_width() {
-                        err.span_suggestion_with_applicability(
-                            expr.span,
-                            &format!("{} in a lossless way", msg),
-                            into_suggestion,
-                            Applicability::MachineApplicable
-                        );
+                        if is_suffixed(expr) {
+                            err.span_suggestion_with_applicability(
+                                expr.span,
+                                &format!(
+                                    "change the type of the numeric literal from `{}` to `{}`",
+                                    checked_ty,
+                                    expected_ty,
+                                ),
+                                suffix_suggestion,
+                                Applicability::MaybeIncorrect,
+                            );
```

Comment:
I feel there's a bit of duplication on these, would it make sense to have `is_suffixed` (with a more accurate name) take `mut err`, `checked_ty` and `expected_ty`, and if it would return true now it would also apply the suggestion, and here you just call it as `if !is_sufficed(expr) {` with the following block (as each one of these has slightly different wording).